### PR TITLE
 set up mixpanel

### PR DESCRIPTION
**Pull request by miliPaoletti**
Setup Mixpanel for tracking events on our dashboard.
Event list: https://www.notion.so/Events-to-Track-with-Mixpanel-fa70b600c5604f84a6840a0baf91910b

### CODE REVIEW

File: packages/ui-helpers/src/__tests__/Header.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  render,
+  renderHook,
+  act,
+} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import { CLICK_EVMOS_LOGO, MixpanelProvider, useTracker } from "tracker";
+import { Header } from "../Header";
+import { CONFIG, TOKEN, eventTriggerByText } from "./testConstants";
+
+describe("Testing for Header", () => {
```
Comment (sandoche) [inline packages/ui-helpers/src/__tests__/Header.spec.tsx:14]: Same comment here

File: packages/ui-helpers/src/__tests__/Header.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    const wrapper = ({ children }: { children: JSX.Element }) => (
+      <MixpanelProvider token={TOKEN} config={CONFIG}>
+        {children}
+      </MixpanelProvider>
+    );
+
+    return wrapper;
+  };
+
+  it("Header in Governance: clicks on Logo and triggers the tracker action", () => {
```
Comment (sandoche) [inline packages/ui-helpers/src/__tests__/Header.spec.tsx:33]: Same comment here

" it("should tigger the tracker action when clicking on ..."

File: packages/ui-helpers/src/__tests__/Navigation.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  render,
+  renderHook,
+  act,
+} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import { CLICK_BACK_TO_MC, MixpanelProvider, useTracker } from "tracker";
+import { Navigation } from "../Navigation";
+import { NAV_TO_MISSION_CONTROL, EVMOS_PAGE_URL } from "constants-helper";
+import { CONFIG, TOKEN, eventTriggerByText } from "./testConstants";
+describe("Testing for Navigation", () => {
```
Comment (sandoche) [inline packages/ui-helpers/src/__tests__/Navigation.spec.tsx:14]: Same comment here

File: packages/ui-helpers/src/__tests__/Navigation.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    const wrapper = ({ children }: { children: JSX.Element }) => (
+      <MixpanelProvider token={TOKEN} config={CONFIG}>
+        {children}
+      </MixpanelProvider>
+    );
+
+    return wrapper;
+  };
+
+  it("Navigation: track back to mission control click", () => {
```
Comment (sandoche) [inline packages/ui-helpers/src/__tests__/Navigation.spec.tsx:33]: Same comment as abode

File: packages/ui-helpers/src/__tests__/Footer.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  TOKEN,
+  eventTriggerByText,
+} from "./testConstants";
+
+describe("Testing for Footer", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("calls Mixpanel functions without token", () => {
```
[the remark this answers is not in the document]
Comment (sandoche) [reply packages/ui-helpers/src/__tests__/Footer.spec.tsx:24]: Also it should not call mixpanel if there is no token no?

File: packages/ui-helpers/src/__tests__/Switch.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  renderHook,
+  act,
+} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import { CLICK_HIDE_ZERO_BALANCE, MixpanelProvider, useTracker } from "tracker";
+import { Switch } from "../Switch";
+import { useState } from "react";
+import { CONFIG, TOKEN, eventTriggerByCheckbox } from "./testConstants";
+
+describe("Testing for Switch", () => {
```
Comment (sandoche) [inline packages/ui-helpers/src/__tests__/Switch.spec.tsx:15]: Same comment

File: packages/ui-helpers/src/__tests__/Switch.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("Switch: track checked status on click", () => {
```
Comment (sandoche) [inline packages/ui-helpers/src/__tests__/Switch.spec.tsx:20]: Same comment here

File: packages/ui-helpers/src/__tests__/testConstants.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+export const CONFIG = { debug: true, ip: false };
+
+export const eventTriggerByText = (expectedText: string) => {
+  const textElement = screen.getByText(expectedText);
+  expect(textElement).toBeInTheDocument();
+  fireEvent.click(textElement);
+};
+
+export const eventTriggerByCheckbox = (checkboxElement: HTMLInputElement) => {
+  fireEvent.click(checkboxElement);
```
Comment (sandoche) [inline packages/ui-helpers/src/__tests__/testConstants.ts:13]: When possible try to use user userEvent,click instead of fireEvent,click

**Review by sandoche — APPROVED** [review]
I approve it but I suggest that we go over the tests together later.

File: packages/ui-helpers/src/__tests__/testConstants.ts (hunk 8 above)
Comment (miliPaoletti) [reply packages/ui-helpers/src/__tests__/testConstants.ts:13]: great! I'll change it on the other ticket :)

File: packages/ui-helpers/src/__tests__/Footer.spec.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (miliPaoletti) [reply packages/ui-helpers/src/__tests__/Footer.spec.tsx]: yeah, exactly. I'll review it in the other ticket

**Pull request merged**